### PR TITLE
Basic Video Functionality with User Interface

### PR DESCRIPTION
**Pull request by nickboucher**
This merge adds basic P2P video chat functionality through the WebRTC protocol. A signaling server is implemented via socket.io websockets, which are able to co-exist with the textroullete socket.io paradigm. STUN and TURN servers are used to bypass network firewall restrictions, and we rely on Google and Xirsys to provide these services.

This merge does not add the end-goal matching behavior for the video chat platform, rather it simply joins all requesting users into a room of their choice for basic P2P video functionality. The desired matching functionality will be incorporated with the next pull request.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,6 +4,10 @@
   "description": "CS96 Dev Team first project -- a text-based communication tool, driven by Node.js",
   "main": "server.js",
   "dependencies": {
+    "getconfig": "2.1.0",
+    "node-uuid": "1.2.0",
+    "socket.io": "1.3.7",
```
Comment (patrickhpan) [inline package.json:9]: Why the downgrade in socket.io version? Did this happen for any reason? If not, we should 

`npm install --save socket.io@latest`

File: sockets.js [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+                    type: type
+                });
+                if (!type) {
+                    client.leave(client.room);
+                    client.room = undefined;
+                }
+            }
+        }
+
+        function join(name, cb) {
```
Comment (patrickhpan) [inline sockets.js:87]: what's the point of `cb`? any way we can convert this to use `Promise` instead to be cleaner?

File: sockets.js [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+        return result;
+    }
+
+    function clientsInRoom(name) {
+        return io.sockets.clients(name).length;
+    }
+
+};
+
+function safeCb(cb) {
```
Comment (patrickhpan) [inline sockets.js:182]: wouldn't be necessary if we can use `Promise`

**Review by patrickhpan — CHANGES_REQUESTED** [review]
looks great if it works! just a couple small changes

File: package.json (hunk 1 above)
Comment (AlexAbes) [reply package.json:9]: We spent a long time trying to make our code work with the new 2.0.4 version, and failed. There are breaking changes between the 2 versions, and it was easier to rectify the changes in our own code than the webRTC library we are using (though we did try it for a long time). We did not find any features in 2.0.4 that we need right now, so this seemed reasonable. Is there anything specific in v2 you need or is it just a general preference to work on the most up-to-date version?

Comment (patrickhpan) [reply package.json:9]: got it. I looked at the diff between the two versions and it looks like it's not super critical to use the latest version, especially if it'll be a time sink in the short term

File: sockets.js (hunk 2 above)
Comment (nickboucher) [reply sockets.js:87]: This code was taken directly from [Signalmaster](https://github.com/andyet/signalmaster) (MIT License), the signaling server designed to work with SimpleWebRTC

**Review by patrickhpan — APPROVED** [review]

**Pull request merged**